### ISSUE TITLE
Huggingface-Aishell get wrong prediction

### ISSUE THREAD

**Issue by Ht-zhang-xianyu** [speechbrain/speechbrain#2549]
### Describe the bug

Use the pretrained from huggingface case in Aishell dataset, but the prediction seems strange 
The script I use is as follows:

`
from speechbrain.inference.ASR import EncoderDecoderASR

asr_model = EncoderDecoderASR.from_hparams(source="speechbrain/asr-transformer-aishell", savedir="pretrained_models/asr-transformer-aishell")
asr_model.transcribe_file("speechbrain/asr-transformer-aishell/example_mandarin2.flac")`

but return the predict token is :
"一 日 一一 一一 一一 六 克一 件 第 一"

### Expected behaviour

expect the correct asr script of wav

**Comment by LudwigFancy** [comment]
i meet the same question, then i use this pretrained model to test another examples which is created by myself, but more errors happened. i don't  understand whether the data need a preprocess when we test?

**Comment by asumagic** [comment]
Seems like `develop` is even more broken due to the `mutihead_attn` rename, resulting in some layer weights not being loaded correctly:

> `loading from pretrained_models/asr-transformer-aishell/asr.ckpt, the object could not use the parameters loaded with the key: 1.decoder.layers.5.mutihead_attn.att.out_proj.bias`

Need to get #2529 to solve this.

That being said, the current issue above is separate.

**Comment by asumagic** [comment]
This is a regression from the switch to SB `1.0.0`:

- `3eb53d5` + `v0.5.11` = `企业 专业 领域 仍是 微软 可以 坚守 的 阵地`
- `7bacef7` + `v0.5.16` = `企业 专业 领域 仍是 微软 可以 坚守 的 阵地`
- **[`0aedced`](https://huggingface.co/speechbrain/asr-transformer-aishell/commit/0aedced492ddde14d828b77c3c3109cadf9f1cde)** + `v1.0.0` = `一 日 一一 一一 一一 六 克一 件 第 一`

Where the short commit hash is the relevant HF repo. The last one seems like a possible culprit, unless it's a bug from elsewhere in `1.0`.

**Comment by asumagic** [comment]
OK, I think I found the issue, but I need to do some more investigation as to the why. The issue seems to be that a causal mask is being applied when it should not. IIRC, something was fixed about causal encoders for transformers. This likely has broken models trained with `causal: True` (the default) that in fact weren't using causality at training time.

The simple fix is to add `causal: False` in the model hyperparameters, and it works with the current unchanged `develop`.
